### ISSUE TITLE
OAuth access token gets shared among all users for provider API requests

### ISSUE THREAD

**Issue by domfarolino** [miguelgrinberg/flask-oauth-example#5]
If someone were to build a basic site off this framework for multiple users and perhaps add an API call in the back that touches an OAuth provider endpoint to get user information, the access token for the last logged in user (with that provider) would be used to make this call so there could potentially be a breach in privacy in trying to request your own information from a provider's API. I faced this when working with this code.

I understand its a small framework primarily demonstrating the usage of OAuth but it seems like a reasonable concern. I can submit a PR to fix it if it will be considered. I have addressed this in my material-message-board repository using a class method and some session information

**Comment by miguelgrinberg** [comment]
I honestly don't see where a token is shared. The object that holds the token is never persisted from one request to the next, as far as I can see. If you wanted to make a call to the provider API you would need to save the user access token (something this example application does not do), and recreate the OAuth session with it when you are ready to make the call. At least that was the intention.

Can you explain better where you see this vulnerability? Or if you already have made code changes to address it, go ahead and send a PR and I'll review it. Thanks.

**Comment by domfarolino** [comment]
Ahh yes, I manually persisted the `oauth_session` variable found in both provider callback functions as `self.oauth_session`. I did this so I didn't have to rebuild it on every provider API request I would make however I overlooked the fact that the persisting would bite me later. As a result I wrote a class method session-based fix in `OAuthSignIn` to help deal with the swapping in and out of user specific access token information.

I should've taken a closer look at your `oauth.py` to realize you did not do this (sorry!)...when writing up this issue I was thinking for some reason the OAuth subclasses were holding and persisting this data.
